src/EditorFeatures/Core/Shared/Extensions/GlyphExtensions.cs

```diff
@@ -1,12 +1,206 @@
 ﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
 
+using System;
 using System.Collections.Immutable;
 using Microsoft.CodeAnalysis.Tags;
+using Microsoft.VisualStudio.Core.Imaging;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
 
 namespace Microsoft.CodeAnalysis.Editor.Shared.Extensions
 {
     internal static class GlyphExtensions
     {
+        public static ImageId GetImageId(this Glyph glyph)
+        {
+            var moniker = glyph.GetImageMoniker();
+            return new ImageId(moniker.Guid, moniker.Id);
+        }
+
+        public static ImageMoniker GetImageMoniker(this Glyph glyph)
```

Comment:
I thought part of the goal was to not have any mention of `ImageMoniker` in EditorFeatures, and to instead directly translate from Glyph to ImageId without the middle step (for Glyph.Assembly, it'd look like `new ImageId(KnownImageIds.ImageCatalogGuid, KnownImageIds.Assembly)`)

Reply:
(And leave the ImageMoniker one in EditorFeatures.WPF for consumers who need actual ImageMonikers)

Reply:
Correct: this needs to become GetImageId() with the really huge switch. Then GetImageMoniker() stays in EditorFeatures.Wpf and just converts from Id to Moniker.